File: windows/codexgauge_windows/file_locations.py
from __future__ import annotations

import os
import shutil
from pathlib import Path


def appdata_directory() -> Path:
    appdata = os.environ.get("APPDATA")
    if appdata:
        return Path(appdata)
    return Path.home() / "AppData" / "Roaming"


APP_SUPPORT_DIRECTORY = appdata_directory() / "CodexControl"
LEGACY_APP_SUPPORT_DIRECTORIES = [
    appdata_directory() / "CodexGauge",
    appdata_directory() / "CodexAccounts",
]
ACCOUNTS_FILE = APP_SUPPORT_DIRECTORY / "accounts.json"
SNAPSHOTS_FILE = APP_SUPPORT_DIRECTORY / "snapshots.json"
MANAGED_HOMES_DIRECTORY = APP_SUPPORT_DIRECTORY / "managed-homes"
AUTH_BACKUPS_DIRECTORY = APP_SUPPORT_DIRECTORY / "auth-backups"
AMBIENT_CODEX_HOME = Path.home() / ".codex"


def ensure_directories() -> None:
    if not APP_SUPPORT_DIRECTORY.exists():
        for legacy_directory in LEGACY_APP_SUPPORT_DIRECTORIES:
            if legacy_directory.exists():
                shutil.move(str(legacy_directory), str(APP_SUPPORT_DIRECTORY))
                break

    APP_SUPPORT_DIRECTORY.mkdir(parents=True, exist_ok=True)
    MANAGED_HOMES_DIRECTORY.mkdir(parents=True, exist_ok=True)
    AUTH_BACKUPS_DIRECTORY.mkdir(parents=True, exist_ok=True)

File: Sources/CodexAccounts/Support/FileLocations.swift
import Foundation

enum FileLocations {
    static var appSupportDirectory: URL {
        let base = FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask).first
            ?? FileManager.default.homeDirectoryForCurrentUser
        return base.appendingPathComponent("CodexGauge", isDirectory: true)
    }

    static var legacyAppSupportDirectory: URL {
        let base = FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask).first
            ?? FileManager.default.homeDirectoryForCurrentUser
        return base.appendingPathComponent("CodexAccounts", isDirectory: true)
    }

    static var accountsFile: URL {
        self.appSupportDirectory.appendingPathComponent("accounts.json", isDirectory: false)
    }

    static var snapshotsFile: URL {
        self.appSupportDirectory.appendingPathComponent("snapshots.json", isDirectory: false)
    }

    static var managedHomesDirectory: URL {
        self.appSupportDirectory.appendingPathComponent("managed-homes", isDirectory: true)
    }

    static var authBackupsDirectory: URL {
        self.appSupportDirectory.appendingPathComponent("auth-backups", isDirectory: true)
    }

    static var ambientCodexHome: URL {
        FileManager.default.homeDirectoryForCurrentUser.appendingPathComponent(".codex", isDirectory: true)
    }

    static func ensureDirectories() throws {
        if !FileManager.default.fileExists(atPath: self.appSupportDirectory.path),
           FileManager.default.fileExists(atPath: self.legacyAppSupportDirectory.path)
        {
            try FileManager.default.moveItem(at: self.legacyAppSupportDirectory, to: self.appSupportDirectory)
        }
        try FileManager.default.createDirectory(at: self.appSupportDirectory, withIntermediateDirectories: true)
        try FileManager.default.createDirectory(at: self.managedHomesDirectory, withIntermediateDirectories: true)
        try FileManager.default.createDirectory(at: self.authBackupsDirectory, withIntermediateDirectories: true)
    }
}
